### ISSUE TITLE
Ownclouds new Content-Security-Policy

### ISSUE THREAD

**Issue by sualko** [jsxc/jsxc#201]
With the release of Owncloud 8.1 the <code>custom_csp_policy</code> was removed and therefore the only way to connect to a bosh server is now via proxy. Maybe someday this function will come back owncloud/core#16927.

However, the major problem is that every application can set is own csp and the appframework has a really restrictive one. So they also block images from <code>data:</code> sources, which blocks user avatars in jsxc. I will try do remove this restriction with a pull request. Stay tuned.

**Comment by sualko** [comment]
Now also the file app sets a very restrictive csp <code>default-src 'self'; script-src 'self' 'unsafe-eval'; style-src 'self' 'unsafe-inline'; frame-src *; img-src *; font-src 'self' data:; media-src *; connect-src *</code>. Therefore a lot of icons are not loaded, because for faster loading I included them in the stylesheet.

@LukasReschke Is there any config option to allow <code>data:</code> as source for images? Or do I need to create a new pr similar to owncloud/core#18096?

**Comment by sualko** [comment]
With oc 9 also video streams are blocked:
```
Refused to load media from "blob:..." because it violates the following Content Security Policy directive: "media-src self"
```
and web worker which are used to generate our DSA key in background:
```
Refused to create a child context containing ... because it violates the following Content Security Policy directive: "default-src none". Note that child-src was not explicitly set, so default-src is used as a fallback.
```

**Comment by LukasReschke** [comment]
https://github.com/owncloud/core/pull/21989 should make you happy.

**Comment by sualko** [comment]
Very happy! :clap: @LukasReschke  Thank you so much. Only one more question: App.php is cached and therefore no dynamic changes are possible. Am I right?

**Comment by LukasReschke** [comment]
No. app.php is evaluated on each request (not entirely correct, there are cases where we don't load it, but they are not relevant for you)

So you can do any fancy stuff you want in there. Just be aware that if your fancy stuff is slow all of ownCloud will be slow :)

**Comment by sualko** [comment]
Great news for most users, because adding a proxy is one more hurdle. Just adding one domain from a app config should not noticeable slow down owncloud, I think.

**Comment by LukasReschke** [comment]
> Great news for most users, because adding a proxy is one more hurdle. Just adding one domain from a app config should not noticeable slow down owncloud, I think.

Yeah. That's fine, we even have some advanced compliance magic in the Enterprise edition that uses this feature. As long as you don't add `sleep(3000)` or so all should be fine. And great app btw. :rocket: :dancers:

**Comment by sualko** [comment]
fix works in 9.0 and above.
